ios/Tests/MappersTests.swift

```diff
@@ -19,4 +19,14 @@ final class MappersTests: XCTestCase {
         XCTAssertEqual(Mappers.mapFromLocationStatus(.notSet), "notSet")
     }
 
+    func testCollectInputsReturnsMapper() {
+        let output: NSDictionary = Mappers.mapFromCollectInputsResults([])
```

Comment:
Hey @bric-stripe, I'm trying to add a test for checking the return data of method  `Mappers.mapFromCollectInputsResults`,  I'm block with a issue that how to mock its input param which is a `CollectInputsResult` array, and the class `CollectInputsResult` and all its subclass can't be directly instantiated. I have no idea how to create or hardcode the input params, could you give me some guidance, thanks.

Reply:
ah, this is because all the public classes are annotated as init and new being unavailable because we want to ensure the instances the SDK receives are generated from the SDK. But this means you can't simply instantiate them for tests either 😞 

for result classes like this we should probably remove that restriction from the native SDK (its only important classes that are also inputs, like PaymentIntent)

the only option I can think of for now is to create mirror protocols that the Mapper class, and any others being tested, use and then the RN swift layer will declare that the SDK's class conforms to that protocol. I'll check out this branch and see if I can add an example commit to help unblock. It's going to be noisy though as we have to declare each param in the protocol 😞 I'll also bring this up with others to see if there are alternatives to consider.

Reply:
Understood! Thanks a lot for your detailed teaching. I'm also going to do some research on that.

Reply:
sorry I didn't get to this yesterday but I just pushed up a commit that does what I was thinking: [5260769](https://github.com/stripe/stripe-terminal-react-native/pull/764/commits/52607692fc610fe97f7bccfded3669a2e18f85c5)

the downside is we'd have to keep the protocols in sync. But since the public classes from the native SDK are fairly stable I don't think that should be too bad. But there may be better alternatives that I'm not aware of.

Reply:
Thanks for your instructive sample codes, but there seems to be some errors😂, could you take a look at it.

Reply:
silly error, I just missed adding a new file. Pushed a new commit that includes that and should fix CI